client/knowledge_base_search/src/containers/SearchContainer/SearchContainer.js

```diff
@@ -42,84 +45,108 @@ class SearchContainer extends Component {
     return `View all ${numPresetQueries.toLocaleString()}+ questions`;
   }
 
-  render() {
+  handleOnQuestionTypeSelect = (e) => {
+    this.setState({ selectedQuestionType: e.target.value });
+  }
+
+  renderQuestionType() {
     const {
-      errorMessage,
       isFetchingQuestions,
-      isFetchingResults,
-      presetQueries,
+      errorMessage,
       offset,
       onOffsetUpdate,
       onQuestionClick,
+      presetQueries,
+      isFetchingResults,
       onViewAllClick
     } = this.props;
 
+    switch (this.state.selectedQuestionType) {
```

Comment:
A switch statement here is fine, but you could bounce it out to a function if you felt the need to "clean up" the render return. Then your return could just be `return( <div>{ this.getSelectContent }</div>);`

I don't see it being a big deal either way though.

Reply:
i agree with this - i will address this refactor in another commit